tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/ConcreteFunction.java

```diff
@@ -154,141 +240,235 @@ public Signature signature() {
-  public Graph graph() {
-    return graph;
+  public Map<String, Operand<?>> call(Ops tf, Map<String, Operand<?>> arguments) {
+    return tf.call(this, arguments);
+  }
+
+  /**
+   * Calls the function in an execution environment, adding it's graph as a function if it isn't already present. Only
+   * works for functions with a single input and output.
+   *
+   * @param tf the scope to call the function in
+   * @param argument the argument to the call
+   * @return the output of the function
+   */
+  public Operand<?> call(Ops tf, Operand<?> argument) {
+    return tf.call(this, argument);
+  }
+
+  /**
+   * Export this function as a saved model.
+   *
+   * <p>This method is convenient shortcut equivalent to
+   * {@code SavedModel.exporter(exportDir).withFunction(this).export()}
+   *
+   * @param exportDir directory where to export the saved model
+   * @throws IOException if saved model or variable state cannot be written on disk
+   */
+  public void save(String exportDir) throws IOException {
+    SavedModelBundle.exporter(exportDir).withFunction(this).export();
   }
 
   @Override
   public void close() {
-    if (ownership != Ownership.NONE) {
-      session.close();
-      if (ownership == Ownership.GRAPH_AND_SESSION) {
-        graph.close();
-      }
-    }
+    scope.close();
   }
 
   @Override
   public String toString() {
     return signature.toString();
   }
 
-  private enum Ownership {
-    GRAPH_AND_SESSION, SESSION_ONLY, NONE;
+  /**
+   * FIXME: This causes native errors when I use it (Linux GPU, 6.1 CC), but I'm leaving it because how to enable XLA
+   * JIT is extremely non-obvious.
+   *
+   * Causes {@code OP_REQUIRES failed at xla_ops.cc:363 : Not found: could not find registered platform with id:
+   * 0x7f75af03e6e8} (it's a warning, but the resulting TF_Status fails).
+   */
+  private void makeJit() {
+    try (PointerScope scope = new PointerScope()) {
+      byte[] bytes = AttrValue.newBuilder().setB(true).build().toByteArray();
+      BytePointer trueValue = new BytePointer(bytes);
+
+      TF_Status status1 = TF_Status.newStatus();
+      TF_FunctionSetAttrValueProto(nativeHandle(), "_XlaMustCompile", trueValue, bytes.length, status1);
+      status1.throwExceptionIfNotOK();
+
+      TF_Status status2 = TF_Status.newStatus();
+      TF_FunctionSetAttrValueProto(nativeHandle(), "_noinline", trueValue, bytes.length, status2);
+      status2.throwExceptionIfNotOK();
+    }
+  }
+
+  TF_Function nativeHandle() {
+    if (nativeHandle.isNull()) {
+      throw new IllegalStateException("Function has been closed");
+    }
+    return nativeHandle;
+  }
+
+  /**
+   * Get the native handle of the function's gradient, so that it can be attached to a Graph.  Not implemented yet.
+   *
+   * TODO implement
+   */
+  TF_Function gradNativeHandle() {
+    return null;
   }
 
-  private final Graph graph;
-  private final Session session;
   private final Signature signature;
-  private final Ownership ownership;
```

Comment:
Why did you remove the ownership part? It is quite important to prevent unexpected behavior when closing a function that comes from a saved model vs. one that has been instantiated manually.